src/VersionParsing.jl

```diff
@@ -76,4 +76,10 @@ function vparse(s_::String)
     end
 end
 
+function _precompile_()
+    ccall(:jl_generating_output, Cint, ()) == 1 || return nothing
+    precompile(Tuple{typeof(VersionParsing.vparse), String})
+end
+_precompile_()
```

Comment:
What's the advantage of this over calling `precompile(...)` unconditionally?

Reply:
I've just seen everyone do it like this so it only runs when the package is precompiling (and not when it is loaded with e.g. `--compiled-modules=no`). But I could just do it like that if you prefer.

Reply:
ccall'ing an undocumented internal function seems like a weird idiom to encourage. 

Why isn't `precompile` just a no-op if `--compiled-modules=no`?

Reply:
I don't know. How do you want to procede here?